hs-bindgen/examples/golden/typedefs.h

```diff
@@ -1,2 +1,5 @@
 typedef int myint;
 typedef int * intptr;
+
+typedef void (*FunctionPointer_Function)(void);
+typedef int (NonFunctionPointer_Function)(int value);
```

Comment:
If you add the following example here:
```c
//! Pointer-to-function type
typedef void (*f1)(void);

typedef void g1(void);
//! Pointer-to-function type, but with one typedef indirection
typedef g1 * g2;

typedef void h1(void);
typedef h1 h2;
//! Pointer-to-function type, but with two typedef indirections
typedef h2 * h3;
```
Then you can see that `getUnderlyingType` works as expected